lib/src/layer/polygon_layer/painter.dart

```diff
@@ -39,42 +39,41 @@ base class _PolygonPainter<R extends Object>
     required LatLng coordinate,
   }) {
     final polygon = projectedPolygon.polygon;
-
-    if (!polygon.boundingBox.contains(coordinate)) return false;
+    if (!polygon.boundingBox.contains(coordinate)) {
+      return false;
+    }
 
     final projectedCoords = getOffsetsXY(
       camera: camera,
       origin: hitTestCameraOrigin,
       points: projectedPolygon.points,
-    ).toList();
+    );
 
     if (projectedCoords.first != projectedCoords.last) {
       projectedCoords.add(projectedCoords.first);
     }
+    final isInPolygon = isPointInPolygon(point, projectedCoords);
 
     final hasHoles = projectedPolygon.holePoints.isNotEmpty;
-    late final List<List<Offset>> projectedHoleCoords;
-    if (hasHoles) {
-      projectedHoleCoords = projectedPolygon.holePoints
-          .map(
-            (points) => getOffsetsXY(
+    final isInHole = hasHoles &&
+        () {
+          for (final points in projectedPolygon.holePoints) {
+            final projectedHoleCoords = getOffsetsXY(
               camera: camera,
               origin: hitTestCameraOrigin,
               points: points,
-            ).toList(),
-          )
-          .toList();
+            );
 
-      if (projectedHoleCoords.firstOrNull != projectedHoleCoords.lastOrNull) {
```

Comment:
I don't think this does what the author intended. This pushes a list and not a coordinate